app/src/main/java/com/example/drawingApp/DialogObject.kt

```diff
@@ -0,0 +1,119 @@
+package com.example.drawingApp
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import android.widget.EditText
+import android.widget.TextView
+import android.widget.Toast
+import androidx.appcompat.app.AlertDialog
+import androidx.core.view.isVisible
+import androidx.core.widget.addTextChangedListener
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.google.android.material.bottomsheet.BottomSheetDialog
+
+object DialogObject {
+
+    fun showLayerAlertDialog(
+        c: Context,
+        model: DrawingViewModel,
+        onDialogSubmission: (String) -> Unit
+    ) {
+        val inflater = LayoutInflater.from(c)
+        val view = inflater.inflate(R.layout.layer_dialog, null)
+        val layerField = view.findViewById<EditText>(R.id.newLayerField)
+        val errorText = view.findViewById<TextView>(R.id.dialogError)
+        val layerNames = mutableListOf<String>()
+        for (x in model.layerNames) layerNames.add(x.text)
+        var layerCounter = 1
+
+
+        //find the first available layer name not taken
+        while (layerNames.contains(
+                c.resources.getString(
+                    R.string.new_layer_hint,
+                    layerCounter
+                )
+            )
+        ) {
+            layerCounter += 1
+        }
+        layerField.setText(c.resources.getString(R.string.new_layer_hint, layerCounter))
+
+        //needs to be after the set text or else the textchanged listener will overwrite the previous text needed for rotations
+        layerField.addTextChangedListener {
+            val fieldString = it.toString().trim()
+            model.currentLayerCreationDialogText = fieldString
+            errorText.setText(R.string.charCountError)
+            errorText.isVisible = fieldString.length > MainActivity.MAX_CHAR_LEN
+        }
+
+        val dialog = AlertDialog.Builder(c).apply {
+            setView(view)
+        }.create()
+
+        view.findViewById<TextView>(R.id.newLayerCancel).setOnClickListener {
+            dialog.dismiss()
+        }
+
+        view.findViewById<TextView>(R.id.newLayerOk).setOnClickListener {
+            val fieldString = layerField.text.toString().trim()
+            if (fieldString.length > MainActivity.MAX_CHAR_LEN) {
+                errorText.visibility = View.VISIBLE
```

Comment:
I would use isVisible instead of .visibility here

Reply:
because of the requirement of the below, it would be weird to use isVisible here and setting visibility there.